ee/server/calendar/google_calendar.go

```diff
@@ -350,6 +348,14 @@ func (c *GoogleCalendar) GetAndUpdateEvent(event *fleet.CalendarEvent, genBodyFn
 		}
 	}
 
+	// If event was deleted/cancelled, we need to stop watching it
+	if !channelStopped {
```

Comment:
Are you sure there are no paths where the function returns and this will not be called even if it's needed?
Maybe consider putting under `defer`?

Reply:
I think it is OK as is -- I reviewed the code. I will rework the code so that we don't stop/create a new channel to watch the calendar if event changes. Maybe that will help with callback latency. That change can go into the next release/patch.